yb-voyager/src/query/queryissue/parser_issue_detector.go

```diff
@@ -1153,13 +1178,81 @@ func (p *ParserIssueDetector) detectTablePKRecommendations(tm *TableMetadata) []
 		}
 	}
 
-	if len(options) > 0 {
-		issues = append(issues, NewMissingPrimaryKeyWhenUniqueNotNullIssue("TABLE", tm.GetObjectName(), options))
+	// Process UNIQUE constraints
+	for _, uniqueCols := range tm.GetUniqueConstraints() {
+		processUniqueColumns(uniqueCols)
+	}
+
+	// Process UNIQUE indexes (column-based only)
+	for _, uniqueCols := range tm.GetUniqueColumnBasedIndexes() {
+		processUniqueColumns(uniqueCols)
+	}
+
+	// Remove duplicates from options using lo.UniqBy
+	uniqueOptions := lo.UniqBy(options, func(option []string) string {
+		return strings.Join(option, ",")
+	})
+
+	if len(uniqueOptions) > 0 {
+		issues = append(issues, NewMissingPrimaryKeyWhenUniqueNotNullIssue("TABLE", tm.GetObjectName(), uniqueOptions))
 	}
 
 	return issues
 }
 
+// getAllPartitionColumnsInHierarchy returns all partition columns down the entire hierarchy
+// for a root-level partitioned table. This includes partition columns from all child tables.
+func (p *ParserIssueDetector) getAllPartitionColumnsInHierarchy(tm *TableMetadata) []string {
+	// Using a map to do automatic deduplication in case if same column is present in multiple child tables
+	allPartitionColumns := make(map[string]bool)
+
+	for _, col := range tm.GetPartitionColumns() {
+		allPartitionColumns[col] = true
+	}
+
+	tableNameOnly := tm.TableName
+	for _, childTM := range p.tablesMetadata {
+		if childTM.PartitionedFrom == tableNameOnly {
+			// Recursively get partition columns from this child and all its descendants
+			childPartitionColumns := p.getAllPartitionColumnsInHierarchy(childTM)
+			for _, col := range childPartitionColumns {
+				allPartitionColumns[col] = true
+			}
+		}
+	}
+
+	// Convert map back to slice
+	result := make([]string, 0, len(allPartitionColumns))
+	for col := range allPartitionColumns {
+		result = append(result, col)
+	}
+
+	return result
+}
+
+// hasAnyRelatedTablePrimaryKey checks if this table or any related table already has a primary key.
+// For regular tables: simply checks if the table itself has a PK.
+// For partitioned tables: checks if any table in the partitioning hierarchy (including descendants) has a PK.
+func (p *ParserIssueDetector) hasAnyRelatedTablePrimaryKey(tm *TableMetadata) bool {
```

Comment:
Same here, you can also utilize the function with. `TableMetadata.HasAnyRelatedTablePrimaryKey`. Why do you need separate function?